graphein/datasets/pdb_data.py

```diff
@@ -0,0 +1,1092 @@
+            == len(df) - num_remaining_rows
+        ), "Duplicate rows found in splits."
+
+        return df_splits
+
+    def split_by_deposition_date(
+        self,
+        df: pd.DataFrame,
+        update: bool = False,
+    ) -> Dict[str, pd.DataFrame]:
+        """Split molecules based on their deposition date.
+
+        :param df: DataFrame containing the molecule sequences to split.
+        :type df: pd.DataFrame
+        :param update: Whether to update the selection to the PDB entries
+            defaults to ``False``.
+        :type update: bool, optional
+
+        :return: A Dictionary of split names mapping to DataFrames of
+            sequence splits based on the sequential time frames given.
+        :rtype: Dict[str, pd.DataFrame]
+        """
+        split_time_frames_provided = self.split_time_frames is not None
+        assert split_time_frames_provided
+
+        # Split sequences
+        log.info(
+            f"Splitting sequences into time frames: \
+                {' '.join([str(f) for f in self.split_time_frames])}"
+        )
+        df_splits = self.split_df_into_time_frames(
+            df, self.splits, self.split_time_frames
+        )
+        log.info("Done splitting sequences")
+
+        # Update splits
+        for split in self.splits:
+            if update:
+                df_split = df_splits[split]
+                if self.df_splits[split] is not None:
+                    self.df_splits[split] = self.merge_df_splits(
+                        self.df_splits[split], df_split
+                    )
+                else:
+                    self.df_splits[split] = df_split
+                df_splits[split] = self.df_splits[split]
+
+        return df_splits
+
+    def filter_by_deposition_date(
+        self, max_deposition_date: np.datetime64, update: bool = False
+    ) -> Union[pd.DataFrame, Dict[str, pd.DataFrame]]:
+        """
+        Select molecules deposited on or before a given date.
+
+        :param max_deposition_date: Maximum deposition date of molecule.
+        :type max_deposition_date: np.datetime64
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: Either a single DataFrame of sequences or a
+            Dictionary of split names mapping to DataFrames of
+            sequences split successively by their deposition date.
+        :rtype: Union[pd.DataFrame, Dict[str, pd.DataFrame]]
+        """
+        # Drop missing deposition dates
+        df = self.df.dropna().loc[
+            self.df.deposition_date < max_deposition_date
+        ]
+        if update:
+            self.df = df
+
+        # Split sequences
+        if self.splits_provided:
+            return self.split_by_deposition_date(df, update)
+
+        return df
+
+    def from_fasta(self, ids: str, filename: str) -> pd.DataFrame:
```

Comment:
This should probably update `self.df`

Reply:
Are you referring to the DataFrame splits after calling `self.split_by_deposition_date()` or something else?

Reply:
The motivation here was that it could act as another entry point for users. Instead of starting with the whole PDB and whittling it down, they could load a FASTA file containing a subset and create a split/subset from that.

Reply:
Maybe I'm misunderstanding the workflow you have in mind with this, but to accomplish this outcome, would the user simply be able to pass in `update=True` to our `cluster()` and `filter_by_deposition_date()` functions?

Reply:
If you are comfortable with implementing this, I will leave this feature to you.